quickwit/quickwit-index-management/src/garbage_collection.rs

```diff
@@ -280,15 +307,16 @@ pub async fn delete_splits_from_storage_and_metastore(
         }
     };
     if !successes.is_empty() {
-        let split_ids: Vec<&str> = successes
+        let split_ids: Vec<String> = successes
             .iter()
-            .map(|split_info| split_info.split_id.as_str())
+            .map(|split_info| split_info.split_id.to_string())
             .collect();
-        let metastore_result = protect_future(
-            progress_opt,
-            metastore.delete_splits(index_uid.clone(), &split_ids),
-        )
-        .await;
+        let delete_splits_request = DeleteSplitsRequest {
+            index_uid: index_uid.to_string(),
+            split_ids: split_ids.clone(),
```

Comment:
Are you sure we need to clone?

Reply:
We can remove the error message, is that what you mean? 
I'm fine with removing it, we return the appropriate metastore error that we could log instead.